### ISSUE TITLE
App suddenly stops retrieving new data and after some times restored connection 

### ISSUE THREAD

**Issue by qwqatest** [Faifly/xDrip#193]
Case #1:
App stops show new data for about half an hour and then display it again after Debug Log opening
Case #2:
App stops retrieving new data at 9:31 (Low Power mode was enabled around that time) and didn't restore cobb\nnection until ~ 15:00 when Low Power mode was disabled and app reopen.
<img width="365" alt="Screenshot at Oct 09 15-06-12 (1)" src="https://user-images.githubusercontent.com/47991168/95726757-01c68000-0c82-11eb-9d92-73877edd88b5.png">
<img width="365" alt="Screenshot at Oct 09 15-06-56 (1)" src="https://user-images.githubusercontent.com/47991168/95726768-02f7ad00-0c82-11eb-99a5-1f2c39d56240.png">
<img width="364" alt="Screenshot at Oct 09 15-11-35 (1)" src="https://user-images.githubusercontent.com/47991168/95726770-03904380-0c82-11eb-8539-70348f898ec9.png">

**Comment by qwqatest** [comment]
Now the app losing connection more often and without any reason - when Low Power mode is off and user near the transmitter
![Screenshot at Nov 17 11-25-08](https://user-images.githubusercontent.com/47991168/99371476-9f3b5200-28c7-11eb-8448-077c560acd03.png)

Connection was restored only after the app relaunch

![Screenshot at Nov 17 11-21-08](https://user-images.githubusercontent.com/47991168/99371570-bd08b700-28c7-11eb-999c-6c6cac48a58b.png)

**Comment by qwqatest** [comment]
@NSFriend Still reproducible. 
After a phone call, I saw the alert that readings were missed during the call. 
![Screenshot at Mar 17 16-46-14](https://user-images.githubusercontent.com/47991168/111486909-91e08800-8740-11eb-9227-334ecac8ae48.png)

The app was in background.
![Screenshot at Mar 17 16-46-22](https://user-images.githubusercontent.com/47991168/111486934-973dd280-8740-11eb-89da-7cb759a20bc8.png)

 I opened it and noticed that at least 2 readings are missing.
![Screenshot at Mar 17 16-46-39](https://user-images.githubusercontent.com/47991168/111486949-99a02c80-8740-11eb-936f-aba33b700233.png)
